### ISSUE TITLE
Nested repeater resets the post query which prevents further referencing of the parent query

### ISSUE THREAD

**Issue by damienlee77** [soflyy/oxygen-bugs-and-features#2261]
The new feature of the advanced query is fantastic but will need some things ironed out. This includes the repeater resetting the query to the page rather than its parent. This prevents a repeater from being nested without causing problems. This makes this issue either a bug, or a feature request depending on if the repeater is intended to be nested.

If you have a repeater to display related posts and those posts contain an ACF relationship field containing a list of further related post items, then you must nest a repeater to show those links. However, the nested repeater resets the query to the page query, so any further fields in the parent query can no longer be pulled.

EXAMPLE:

I have a site of product reviews. I have created comparison posts also which each have two product reviews linked with an ACF post object. On the comparison page, I'm pulling the ACF field information from these two reviews for visual comparison. However one of the fields in the product reviews is an ACF relationship field which displays ingredients which are another post type.

I have a repeater that queries each of the two reviews for comparison. I then have a nested repeater query that pulls the related ingredients for that product review. This repeater resets the query to the page query and prevents me from further referencing the parent query of the product review.

**Comment by KittenCodes** [comment]
Replicated & reported internally.

**Comment by damienlee77** [comment]
Yes it replicates in a sandbox site and I have provided a link here: http://hilarious-jaguar.w5.wpsandbox.pro/compare-author/j-k-rowling-vs-j-r-r-tolkien/ to the page which shows how the problem occurs. I've tried to make it as simple as possible. Please let me know if you cant see what's going on.

Cheers,
Damien

**Comment by damienlee77** [comment]
> @damienlee77 I've been able to replicate this issue in the builder only; I can't seem to replicate it on the front-end of the site.
> 
> Are you having problems in the builder, on the front-end, or both? Are you able to reproduce this on a Sandbox site please? https://oxygenbuilder.com/try.

Yes it replicates in a sandbox site and I have provided a link here: http://hilarious-jaguar.w5.wpsandbox.pro/compare-author/j-k-rowling-vs-j-r-r-tolkien/ to the page which shows how the problem occurs. I've tried to make it as simple as possible. Please let me know if you cant see what's going on.

Cheers,
Damien

**Comment by KittenCodes** [comment]
@damienlee77 That's a great example, thank you very much. I will report this internally.

**Comment by loiceoka** [comment]
I have the same issue, similar level of nesting.

**Comment by clockwood** [comment]
Any progress on this?

**Comment by albertvisuals** [comment]
I need this to work on a page. Any news if this will be fixed in 4.0? Is there a work-around?
